dashboard/src/contexts/SettingsEffectsContext.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+
+import React, { createContext, useContext, useCallback, useRef } from 'react';
+import type { UserSettingsUpdate } from '@/entities/userSettings';
+
+type SettingsEffect = {
+  apply: (settings: UserSettingsUpdate) => void;
+};
```

Comment:
Usually with event emitters they are defined as dispatchers, so I suggest making the type of SettingsEffect a `Dispatch<SettingsEffect>`.

```suggestion
type SettingsEffect = Dispatch<SettingsEffect>;
```